feat: added support for create server API and updated Server data model.

sanchitrk opened the pull request:
Added support for create server API as per [docs](https://postmarkapp.com/developer/api/servers-api#create-server) and updated server data model struct.

mrz1836 commented:
@sanchitrk thanks for the contribution. Please review the linting and tests above, some are not passing.

sanchitrk commented:
@mrz1836 thanks for reviewing. Yes there was some issues with linting and example file. I fixed the linting by locally running the `make` command and removed the example file.

mrz1836 reviewed (approved):
LGTM
